domains/auth/Auth.API/API/Controllers/LoginController.cs

```diff
@@ -1,17 +1,40 @@
 using System.ComponentModel.DataAnnotations;
+using System.IdentityModel.Tokens.Jwt;
+using System.Text.Json;
+using API.Controllers.dto;
+using API.Errors;
 using API.Models;
+using API.Orchestrator;
+using API.Repository;
 using API.Services.OidcProviders;
+using API.Utilities;
+using EnergyOriginEventStore.EventStore;
+using FluentValidation;
 using Microsoft.AspNetCore.Mvc;
 
 namespace API.Controllers;
 
 public class LoginController : ControllerBase
 {
-    readonly IOidcService oidcService;
+    private readonly IOidcService oidcService;
+    private readonly IValidator<OidcCallbackParams> validator;
+    private readonly ICryptographyFactory tokenCryptography;
+    private readonly ICryptographyFactory stateCryptography;
+    private readonly IEventStore eventStore;
+    private readonly IUserStorage userStorage;
+    private readonly ICompanyStorage companyStorage;
+    private readonly IOrchestrator orchestrator;
 
-    public LoginController(IOidcService oidcService)
+    public LoginController(IOidcService oidcService, IValidator<OidcCallbackParams> validator, ICryptographyFactory tokenCryptography, ICryptographyFactory stateCryptography, IEventStore eventStore, IUserStorage userStorage, ICompanyStorage companyStorage, IOrchestrator orchestrator)
     {
         this.oidcService = oidcService;
+        this.validator = validator;
+        this.eventStore = eventStore;
+        this.userStorage = userStorage;
+        this.companyStorage = companyStorage;
+        this.orchestrator = orchestrator;
+        this.tokenCryptography = tokenCryptography;
```

Comment:
I believe this constructor could be cleaned when you are using Action Dependency Injection in `CallBackAsync`